### ISSUE TITLE
syncthing can't start because it can't find its binary

### ISSUE THREAD

**Issue by devhell** [NixOS/nixpkgs#9596]
Something must have changed upstream because I get this now in my log files:
```
Failed at step EXEC spawning /nix/store/5s3k9550cai96ixxarbf57yr16xgyzzv-go1.5-syncthing-0.11.22/bin/syncthing: No such file or directory
```
Building `syncthing` from master then shows a directory structure as follows (shortened):

```
result
`-- share
    `-- go
        |-- pkg
        |   `-- linux_amd64
        |       `-- github.com
        |           `-- syncthing
        |               `-- syncthing
        |                   `-- lib
        `-- src
            `-- github.com
                `-- syncthing
                    `-- syncthing
                        |-- [ ... ]
                        [ ... ]
```

It looks like there's a good chunk of stuff missing. :-/

/cc @ctheune @matejc

**Comment by ctheune** [comment]
Syncthing isn't supposed to build with go 1.5 at this point AFAICT. I didn't do the update to .22 but when I tried to prepare .23 today I noticed that the default changed and started using 1.4 explicitly.

Unfortunately there's other stuff (psutils) broken at the moment for me on OS X so I wasn't able to wrap this up today.

**Comment by devhell** [comment]
@ctheune No problem, thanks for looking into this!

**Comment by wkennington** [comment]
It's split into two derivations now, you want the bin one. Sorry I didn't
track down all of these cases, anything that uses buildGoPackage is now
autosplit.

**Comment by devhell** [comment]
@wkennington Sorry, I'm not sure I understand.

**Comment by wkennington** [comment]
pkgs.syncthing.bin contains the syncthing binary, not pkgs.syncthing which
is actually now pkgs.syncthing.dev and contains source and object files.

**Comment by devhell** [comment]
@wkennington Thanks! So then only the module should be updated to reflect this, yes?

**Comment by wkennington** [comment]
You can just change top-level/all-packages to look like vault or consul or
any other of the canonical go examples I've added. They make sure the bin
derivation is the default and only the bin output is accessible to
environment functions.

**Comment by devhell** [comment]
@wkennington I might be an idiot, but `syncthing` isn't in `goPackages`.

**Comment by wkennington** [comment]
Yeah just exchange that part with a callPackage. I'll fix this when I land.

**Comment by devhell** [comment]
@wkennington Thanks!

**Comment by wkennington** [comment]
Should be fixed by 4fecf15b4b09c627f7dfb2792fe202f49b73f90f

**Comment by devhell** [comment]
@wkennington Thank you, much appreciated. ;-)
